### DISCUSSION TITLE
Custom attributes to Vertex and Edge

### DISCUSSION THREAD

**Discussion by xs2tarunkukreja** [vesoft-inc/nebula#5508] [Q&A]
My application have few entities with some fixed set of properties. But user can add any custom property as well. Is such scenario supported by Nebula Graph? If yes, then how?

**Comment by QingZ11** [comment]
If I understand correctly, your requirement can be achieved by adding or removing the properties of the tag/edge? If yes, you can check out the 'alter' command, which provides powerful options to modify tag/edge properties. Here is the documentation: <https://docs.nebula-graph.io/3.4.1/3.ngql-guide/10.tag-statements/3.alter-tag/>. 

Please ensure that your user has the relevant operation permissions, as outlined in the permission documentation. Here is the documentation: <https://docs.nebula-graph.io/3.4.1/7.data-security/1.authentication/3.role-list/>.

**Reply by xs2tarunkukreja** [reply]
Thanks for response.

Yeah alter tag can help us but not fully.

Currently we are using Neo4J. We have Vertex for which 4 properties are fixed. Others user can add anytime just mention like key/values pair. So, if we go with alter command we have to keep altering tags based on users request.  

Is there any other way?

**Reply by xtcyclist** [reply]
You have to alter properties via alterting tags. But you can have multiple tags for a single vertex.

You could try creating some tags for user-defined properties and grant users with the privileges to alter them (adding, removing properties), while preserving some fixed tags with fixed properties and not grating users to alter them.

**Reply by xs2tarunkukreja** [reply]
Few doubt - 
Is there be any performance impact if I will alter tag? In other word, is alter tag command initiate some job? 

Is time taken to execute the alter tag operation depends on number of vertexes having the same tag?
